web/src/main/java/org/openmrs/web/controller/form/FormResourceController.java

```diff
@@ -75,6 +82,12 @@ public String handleAddFormResource(@ModelAttribute("resource") FormResource res
 			Object value = WebAttributeUtil.getValue(request, resource, "resourceValue");
 			resource.setValue(value);
 		}
+		catch (InvalidCustomValueException ex) {
+			HttpSession httpSession = request.getSession();
+			httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, ex.getMessage());
+			log.error("Error while adding form resource", ex);
+			return "redirect:formResources.form?formId=" + resource.getForm().getId();
```

Comment:
We don't usually _redirect_ on a failed post, do we? Normally I think we'd redisplay the form. (Won't this lose the context of the page the user was on?) Also, this is not going to show any specific error message to the end user.